Add release notes for v20240827 (#4517)
Azure/AKS: CHANGELOG.md

# Azure Kubernetes Service Changelog

## Release 2024-08-27

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as v20240827.

### Announcements

* 

### Release Notes

* Features:

>   * CBL-Mariner 1.0 is end of life, creation of new nodepools with OSSKU cblmariner is disabled.

Reviewer: Docs link pls.

* Bug fixes:
  * Fix an Azure NPM issue that user could meet unexpected connectivity for Pods on the Node when editing a NetworkPolicy with a CIDR "except" field.
  * For AGC Windows clusters, fix customenv-file from C:\\k\\akscustom.json to C:\\k\\azurestackcloud.json so that aks-secret-store-provider-azure-windows pod could come up.

>   * Fix a bug where users could not stop/start AgentPools using the PowerState field in the 2020-09-01, 2020-11-01, 2020-12-01, 2021-02-01, 2021-03-01 API versions.

Reviewer: WHerever possible pls capture the GH link of the issue / bug .

  * Fix bug to block non-VMSS (VirtualMachineScaleSets) agent pools in the Automatic SKU validation process.

* Behavior change:
  * [Trusted launch for Azure Kubernetes Service](https://learn.microsoft.com/en-us/azure/aks/use-trusted-launch) user will receive Trusted Launch image upon creating a nodepool with vTPM or secure boot enabled.
  * For non-host network pods running on AKS nodes, they cannot access wireserver(168.63.129.16) port 32526.
  * When [API Server VNet Integration (Preview)](https://learn.microsoft.com/en-us/azure/aks/api-server-vnet-integration) is enabled, all in-cluster traffic from pod to apiserver won't go through ILB.
  * When deploying an [AKS Automatic (preview)](https://learn.microsoft.com/en-us/azure/aks/intro-aks-automatic) cluster, user do not need to register extra feature flags for related preview features, such as APIServerVnetIntegration, NRGLockdown, NodeAutoProvisioning, and Safeguards.
  * For [AKS Automatic (preview)](https://learn.microsoft.com/en-us/azure/aks/intro-aks-automatic) cluster, creating non-VMSS (VirtualMachineScaleSets) agent pools are blocked.

>   * Allow creation of AgentPools without Count field specified if autoscaler enabled.

Reviewer: Any context on this by way of docs would be helpful as to why this change?

Author: Added a closed bug link and move this to bug fix

# Azure Kubernetes Service Changelog

## Release 2024-08-27

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as v20240827.

### Announcements

> * AKS version 1.27 is now deprecated, onboard to LTS if you still need to operate on 1.27.

Reviewer 2: can we add a docs link on switching to LTS?

> * [Trusted launch for AKS](https://learn.microsoft.com/en-us/azure/aks/use-trusted-launch) is now GA.

Reviewer 2: ```suggestion
* [Trusted launch for AKS](https://learn.microsoft.com/azure/aks/use-trusted-launch) is now generally available.
```

Reviewer 2: also this line needs to go under Release Notes -> Features

* The attestation report for [CIS Kubernetes V1.9.0 Benchmark](https://learn.microsoft.com/azure/aks/cis-kubernetes) is published which covers AKS 1.27.x through AKS 1.29.x.

### Release Notes

* Bug fixes:
  * Fix an Azure NPM issue that user could meet unexpected connectivity for Pods on the Node when editing a NetworkPolicy with a CIDR "except" field.
  * Fix a bug where users could not stop/start AgentPools using the PowerState field in the 2020-09-01, 2020-11-01, 2020-12-01, 2021-02-01, 2021-03-01 API versions.
  * Fix bug to block non-VMSS (VirtualMachineScaleSets) agent pools in the Automatic SKU validation process.
  * Fix bug to ensure correct default network plugin settings for Kubernetes clusters using VMAS.
  * Fix bug for intermittent precondition failures when applying an AKS Bicep deployment on the pod subnet delegation.
  * Fix bug of public IP on VMSS dropped after upgrade node image or reset service principal operation.
  * Fix bug (#4282)[https://github.com/Azure/AKS/issues/4282] to remove duplicated toleration from Calico components.

>   * Fix bug to ensure AnnotationControlled is correctly populated by default when creating AKS clusters with app routing enabled, and to ensureAnnotationControlled is an accepted value for the default nginx ingress controller config for AKS clusters with K8s versions <1.30.

Reviewer 2: ```suggestion
  * Fix bug to ensure `AnnotationControlled` is correctly populated by default when creating AKS clusters with app routing enabled, and to ensure `AnnotationControlled` is an accepted value for the default nginx ingress controller config for AKS clusters with K8s versions <1.30.
```

Reviewer 2: can we link doc on `AnnotationControlled` to provide readers context? cc: @sabbour

Reviewer 3: @dyu1208 can we link to this https://learn.microsoft.com/en-us/azure/aks/app-routing-nginx-configuration?tabs=bicep#control-the-default-nginx-ingress-controller-configuration

>   * Fix bug for [karpenter](https://github.com/Azure/karpenter-poc/issues/639).

Reviewer 3: This links to a private repository. Can we document here what the bug was and remove the link? Also the linked issue still seems to be open, was it fixed?

Author: Confirmed with feature owner, removed this line.

* Behavior change:
  * Cluster Stop operations will be blocked if the customer has any illegal webhooks:
    * If APIGroups is wildcard, and Resources is in the criticalResources or wildcard, the webhook will be blocked.

>     * If Resources is wildcard, and APIGroups is in the criticalApiGroups or wildcard, the webhook will be blocked.

Reviewer 2: Have we documented this?

Author: Confirmed with feature owner, the toggle has not been turn on, so removed this change.

  * For non-host network pods running on AKS nodes, they cannot access wireserver(168.63.129.16) port 32526. Before this change user cannot access wireserver port 80, but port 32526 is accessible.

>   * When deploying an [AKS Automatic (preview)](https://learn.microsoft.com/en-us/azure/aks/intro-aks-automatic) cluster, user do not need to register extra feature flags for related preview features, such as APIServerVnetIntegration, NRGLockdown, NodeAutoProvisioning, and Safeguards.

Reviewer 2: ```suggestion
  * When deploying an [AKS Automatic (preview)](https://learn.microsoft.com/azure/aks/intro-aks-automatic) cluster, user do not need to register extra feature flags for related preview features, such as APIServerVnetIntegration, NRGLockdown, NodeAutoProvisioning, and Safeguards.
```

  * Allow creation of AgentPools without Count field specified if autoscaler enabled.
  * CBL-Mariner 1.0 is end of life, creation of new nodepools with OSSKU cblmariner is disabled.
  * Azure Linux 2.0 on Kubernetes version 1.30 LTS will be blocked until Azure Linux 3.0 is released and supported.

>   * KMS v1 is blocked for AKS cluster with K8s version >=1.31.

Reviewer 2: link to relevant doc?

Author: Synced with owner to remove this line as the impact is extremely small

>   * [AGIC addon](https://learn.microsoft.com/en-us/azure/application-gateway/ingress-controller-overview) has been assigned the network contributor role.

Reviewer 2: ```suggestion
  * [Application Gateway Ingress Controller addon](https://learn.microsoft.com/azure/application-gateway/ingress-controller-overview) has been assigned the network contributor role.
```

>   * [AGIC addon](https://learn.microsoft.com/en-us/azure/application-gateway/ingress-controller-overview) has been assigned the network contributor role.

Reviewer 3: Has been assigned where? The [docs](https://learn.microsoft.com/en-us/azure/application-gateway/tutorial-ingress-controller-add-on-new) still say:
> Please ensure the identity used by AGIC has the Microsoft.Network/virtualNetworks/subnets/join/action permission delegated to the subnet Application Gateway is deployed into. If a custom role is not defined with this permission, you may use the built-in Network Contributor role, which contains the Microsoft.Network/virtualNetworks/subnets/join/action permission.

Author: Asked the owner to update the doc

* Component updates:
  * AKS Ubuntu 22.04 image has been updated to [AKSUbuntu-202408.27.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSUbuntu/gen1/2204containerd/202408.27.0.txt).

>   * Azure Linux image has been updated to [AzureLinux-202408.27.0](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSCBLMarinerV2/gen2/202408.27.0.txt). 

Reviewer 2: we need to copy these vhd notes into vhd-notes folder under the AKS repo (last release notes for reference) and then do relative reference

  * Azure Blob Disk CSI driver has been upgraded to [v1.24.3][https://github.com/kubernetes-sigs/blob-csi-driver/releases/tag/v1.24.3] on AKS 1.30, [v1.23.7](https://github.com/kubernetes-sigs/blob-csi-driver/releases/tag/v1.23.7) on AKS 1.29 and 1.28.
  * Azure File CSI driver has been upgraded to [v1.30.5](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.30.5) on AKS 1.30 and 1.29, [v1.29.7](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.29.7) on AKS 1.28.
  * AKS Windows Server 2019 image has been updated to [AKSWindows-2019-17763.6189.240814](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2019-containerd/17763.6189.240814.txt).
  * AKS Windows Server 2022 image has been updated to [AKSWindows-2022-20348.2655.240814](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2022-containerd/20348.2655.240814.txt).

>   * AKS Windows Server 23H2 image has been updated to [AKSWindows-23H2-25398.1085.240814](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/23H2/25398.1085.240814.txt).

Reviewer 2: same as earlier comment. these vhd-notes need to be copied into AKS repo

  * [AGIC addon image](https://learn.microsoft.com/en-us/azure/application-gateway/ingress-controller-overview) has been updated to [v1.7.5](https://github.com/Azure/application-gateway-kubernetes-ingress/releases/tag/1.7.5).
  * Retina Enterprise and Operator image has been updated to [v0.0.9](https://github.com/azure-networking/retina-enterprise/releases/tag/v0.0.9).
  * azure-cloud-controller-manager has been updated to version [v1.30.5](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.30.5), [v1.29.9](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.9), [v1.28.11](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.11), [v1.27.19](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.19).
  * KEDA addon has been updated to [v2.14.1](https://github.com/kedacore/keda/releases/tag/v2.14.1) for Kubernetes = 1.30.
  * Azure Policy addon has been updated to [v1.7.0](https://learn.microsoft.com/en-us/azure/governance/policy/concepts/policy-for-kubernetes#170).

>   * Istio-based service mesh add-on revision asm-1-20 has been upgraded to patch [v1.20.8](https://istio.io/latest/news/releases/1.20.x/announcing-1.20.8/), revision asm-1-21 has been upgraded to patch [v1.21.5](https://istio.io/latest/news/releases/1.21.x/announcing-1.21.5/), and revision asm-1-22 has been upgraded to patch [v1.22.3](https://istio.io/latest/news/releases/1.22.x/announcing-1.22.3/). Users can restart the workload pods to trigger re-injection of the newer patch version of istio-proxy. More information can be found [here](https://learn.microsoft.com/en-us/azure/aks/istio-upgrade#patch-version-upgrade).

Reviewer 3: > revision asm-1-22 has been upgraded to patch [v1.22.3]

Broken link to release notes?

# Azure Kubernetes Service Changelog

## Release 2024-08-27

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as v20240827.

### Announcements

* AKS version 1.27 is now deprecated, onboard to LTS if you still need to operate on 1.27.

> * [Trusted launch for AKS](https://learn.microsoft.com/en-us/azure/aks/use-trusted-launch) is now GA. 

Reviewer 4: This would be "Features" under the release notes. Also please add the GA doc link.

### Release Notes

* Bug fixes:
  * Fix an Azure NPM issue that user could meet unexpected connectivity for Pods on the Node when editing a NetworkPolicy with a CIDR "except" field.

>   * For AGC Windows clusters, fix customenv-file from C:\\k\\akscustom.json to C:\\k\\azurestackcloud.json so that aks-secret-store-provider-azure-windows pod could come up.

Reviewer 4: Implementation details in the message. And if AGC likely not relevant/public

Author: Remove it since it is a AGC fix

  * Fix bug for intermittent precondition failures when applying an AKS Bicep deployment on the pod subnet delegation.
  * Fix bug of public IP on VMSS dropped after upgrade node image or reset service principal operation.
  * Fix bug (#4282)[https://github.com/Azure/AKS/issues/4282] to remove duplicated toleration from Calico components.
  * Fix bug to ensure AnnotationControlled is correctly populated by default when creating AKS clusters with app routing enabled, and to ensureAnnotationControlled is an accepted value for the default nginx ingress controller config for AKS clusters with K8s versions <1.30.
  * Fix bug for [karpenter](https://github.com/Azure/karpenter-poc/issues/639).![image](https://github.com/user-attachments/assets/0957724f-dc1e-4dc0-8864-4ccca8cf61b1)

* Behavior change:
  * Cluster Stop operations will be blocked if the customer has any illegal webhooks:
    * If APIGroups is wildcard, and Resources is in the criticalResources or wildcard, the webhook will be blocked.
    * If Resources is wildcard, and APIGroups is in the criticalApiGroups or wildcard, the webhook will be blocked.

>   * [Trusted launch for Azure Kubernetes Service](https://learn.microsoft.com/en-us/azure/aks/use-trusted-launch) user will receive Trusted Launch image upon creating a nodepool with vTPM or secure boot enabled.

Reviewer 4: What's the behavior change here?

>   * For non-host network pods running on AKS nodes, they cannot access wireserver(168.63.129.16) port 32526.

Reviewer 4: Did they used to? Not clear in the text

Author: Added the previous experience

Reviewer: Docs capturing behavior need not be updated right away but a link here and asking Net PM to do it later is ok too.

  * When deploying an [AKS Automatic (preview)](https://learn.microsoft.com/en-us/azure/aks/intro-aks-automatic) cluster, user do not need to register extra feature flags for related preview features, such as APIServerVnetIntegration, NRGLockdown, NodeAutoProvisioning, and Safeguards.

>   * For [AKS Automatic (preview)](https://learn.microsoft.com/en-us/azure/aks/intro-aks-automatic) cluster, creating non-VMSS (VirtualMachineScaleSets) agent pools are blocked.

Reviewer 4: Isn't this a duplicate from the bug fix above? Also isn't all nodepool creation blocked since this is automatic/karpenter. (hence the bug fix to correct the new VM-Pools)

Author: removed

  * Allow creation of AgentPools without Count field specified if autoscaler enabled.
  * CBL-Mariner 1.0 is end of life, creation of new nodepools with OSSKU cblmariner is disabled.
  * Azure Linux 2.0 on Kubernetes version 1.30 LTS will be blocked until Azure Linux 3.0 is released and supported.
  * KMS v1 is blocked for AKS cluster with K8s version >=1.31.

>   * AGIC addon has been assigned the network contributor role.

Reviewer 4: Please link the respective doc to distinguish AGiC from agc

> * Component updates:

Reviewer 4: No Ubuntu and az Linux images?

>   * Azure Linux 2.0 on Kubernetes version 1.30 LTS will be blocked until Azure Linux 3.0 is released and supported.

Reviewer 4: Please remove this for now. Cc @justindavies @kaarthis

  * Azure Disk CSI driver has been upgraded to [v1.30.3](https://github.com/kubernetes-sigs/azuredisk-csi-driver/releases/tag/v1.30.3) on AKS 1.30, [V1.29.8](https://github.com/kubernetes-sigs/azuredisk-csi-driver/releases/tag/v1.29.8) on AKS 1.28, [1.28.1](https://github.com/kubernetes-sigs/azuredisk-csi-driver/releases/tag/v1.28.10) on AKS 1.27.
  * Azure Blob Disk CSI driver has been upgraded to [v1.24.3][https://github.com/kubernetes-sigs/blob-csi-driver/releases/tag/v1.24.3] on AKS 1.30, [v1.23.7](https://github.com/kubernetes-sigs/blob-csi-driver/releases/tag/v1.23.7) on AKS 1.29 and 1.28.
  * Azure File CSI driver has been upgraded to [v1.30.5](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.30.5) on AKS 1.30 and 1.29, [v1.29.7](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.29.7) on AKS 1.28.
  * AKS Windows Server 2019 image has been updated to [AKSWindows-2019-17763.6189.240814](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2019-containerd/17763.6189.240814.txt).

>   * AKS Windows Server 23H2 image has been updated to [AKSWindows-2022-20348.2655.240814](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/2022-containerd/20348.2655.240814.txt).

Reviewer 4: This is AKS Windows 2022?

>   * AKS Windows Server 23H2 image has been updated to [AKSWindows-23H2-25398.1085.240814](https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSWindows/23H2/25398.1085.240814.txt).

Reviewer 4: Is this actually out?

Reviewer 4: Where's the announcement? How do I use this image? What is it?

Author: Since there is no announcement yet, I'm going to remove it for current notes

# Azure Kubernetes Service Changelog

## Release 2024-08-27

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as v20240827.

### Announcements

> * AKS version 1.27 is now deprecated, onboard to [Long-term support for AKS versions](https://learn.microsoft.com/en-us/azure/aks/long-term-support) if you still need to operate on 1.27.

Reviewer 2: ```suggestion
* AKS version 1.27 is now deprecated. Enable [long-term support for AKS versions](https://learn.microsoft.com/azure/aks/long-term-support) if you still need to operate on 1.27.
```

> * [Trusted launch for AKS](https://learn.microsoft.com/azure/aks/use-trusted-launch) is now generally available.

Reviewer 2: This line needs to be removed (duplicate of L20, which is to be retained)

  * **KEDA 2.15** for Kubernetes >=1.31: The removal of [Pod Identity support](https://github.com/kedacore/keda/issues/5035). If you use pod identity, we recommend you move over to [workload identity for your authentication](https://learn.microsoft.com/azure/aks/keda-workload-identity). 
  * **KEDA 2.14** for Kubernetes = 1.30: The removal of [Azure Data Explorer 'metadata.clientSecret' as it was not safe for managing secrets](https://github.com/kedacore/keda/issues/4514).
  * **KEDA 2.14** for Kubernetes = 1.30: Removal of the [deprecated metricName from trigger metadata section](https://github.com/kedacore/keda/issues/4240). The two impacted Azure Scalers are Azure Blob Scaler and Azure Log Analytics Scaler. If you are using `metricName` today, please move `metricName` outside of trigger metadata section  to`trigger.name` in the trigger section to optionally name your trigger. To view an example of what this would look like, please view the open [GitHub issue](https://github.com/Azure/AKS/issues/4471).

### Release Notes

* Features:
  * [Trusted launch for AKS](https://learn.microsoft.com/azure/aks/use-trusted-launch) is now generally available.

>   * Existing Linux node pools can now be updated to enable or disable Federal Information Process Standard (FIPS). See [aka.ms/aks/updatefips](aka.ms/aks/updatefips) for more information.

Reviewer 2: ```suggestion
  * Existing Linux node pools can now be updated to enable or disable Federal Information Process Standard (FIPS). See [documentation](aka.ms/aks/updatefips) for more information.
```

  * Azure File CSI driver has been upgraded to [v1.30.5](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.30.5) on AKS 1.30 and 1.29, [v1.29.7](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.29.7) on AKS 1.28.
  * AKS Windows Server 2019 image has been updated to [AKSWindows-2019-17763.6189.240814](vhd-notes/AKSWindows/2019/17763.6189.240814.txt).
  * AKS Windows Server 2022 image has been updated to [AKSWindows-2022-20348.2655.240814](vhd-notes/AKSWindows/2022/20348.2655.240814.txt).
  * AKS Windows Server 23H2 image has been updated to [AKSWindows-23H2-25398.1085.240814](vhd-notes/AKSWindows/23H2/25398.1085.240814.txt).
  * AKS App Routing operator image has been updated to [v0.2.3-patch-2](https://github.com/Azure/aks-app-routing-operator/releases/tag/v0.2.3-patch-2) for AKS cluster with K8s versions >=1.30, [v0.2.1-patch-4](https://github.com/Azure/aks-app-routing-operator/releases/tag/v0.2.1-patch-4) for AKS cluster with K8s versions <1.30 to address CVEs.
  * Windows containerd has been updated to [v1.7.20](https://github.com/containerd/containerd/releases/tag/v1.7.20) in AKS cluster with K8s versions >= v1.28.

>   * Kubernetes Secrets Store CSI Driver has been updated to [v1.5.3](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.5.3).

Reviewer 2: this line is not accurate. the next line is correct. secrets store CSI driver is updated to 1.4.4. this line needs to be removed

  * Kubernetes Secrets Store CSI Driver has been updated to [v1.4.4](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.4.4) and Azure Key Vault Provider for Secrets Store CSI Driver to [v1.5.3](https://github.com/Azure/secrets-store-csi-driver-provider-azure/releases/tag/v1.5.3)

>   * [AGIC addon image](https://learn.microsoft.com/en-us/azure/application-gateway/ingress-controller-overview) has been updated to [v1.7.5](https://github.com/Azure/application-gateway-kubernetes-ingress/releases/tag/1.7.5).

Reviewer 2: ```suggestion
  * [Application Gateway Ingress Controller add-on](https://learn.microsoft.com/azure/application-gateway/ingress-controller-overview) image has been updated to [v1.7.5](https://github.com/Azure/application-gateway-kubernetes-ingress/releases/tag/1.7.5).
```

  * Retina Enterprise and Operator image has been updated to [v0.0.9](https://github.com/azure-networking/retina-enterprise/releases/tag/v0.0.9).
  * azure-cloud-controller-manager has been updated to version [v1.30.5](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.30.5), [v1.29.9](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.9), [v1.28.11](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.11), [v1.27.19](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.19).
  * KEDA addon has been updated to [v2.14.1](https://github.com/kedacore/keda/releases/tag/v2.14.1) for Kubernetes = 1.30.

>   * Azure Policy addon has been updated to [v1.7.0](https://learn.microsoft.com/en-us/azure/governance/policy/concepts/policy-for-kubernetes#170).

Reviewer 2: ```suggestion
  * Azure Policy addon has been updated to [v1.7.0](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#170).
```

>   * Istio-based service mesh add-on revision asm-1-20 has been upgraded to patch [v1.20.8](https://istio.io/latest/news/releases/1.20.x/announcing-1.20.8/), revision asm-1-21 has been upgraded to patch [v1.21.5](https://istio.io/latest/news/releases/1.21.x/announcing-1.21.5), and revision asm-1-22 has been upgraded to patch [v1.22.3](https://istio.io/latest/news/releases/1.22.x/announcing-1.22.3). Users can restart the workload pods to trigger re-injection of the newer patch version of istio-proxy. More information can be found [here](https://learn.microsoft.com/en-us/azure/aks/istio-upgrade#patch-version-upgrade).

Reviewer 2: ```suggestion
  * Istio-based service mesh add-on revision asm-1-20 has been upgraded to patch [v1.20.8](https://istio.io/latest/news/releases/1.20.x/announcing-1.20.8/), revision asm-1-21 has been upgraded to patch [v1.21.5](https://istio.io/latest/news/releases/1.21.x/announcing-1.21.5), and revision asm-1-22 has been upgraded to patch [v1.22.3](https://istio.io/latest/news/releases/1.22.x/announcing-1.22.3). Users can restart the workload pods to trigger re-injection of the newer patch version of istio-proxy. More information can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade#patch-version-upgrade).
```

### Announcements

> 

Reviewer 5: Please help add: 

[CIS Kubernetes V1.9.0 Benchmark](https://learn.microsoft.com/azure/aks/cis-kubernetes) is published which covers AKS 1.27.x through AKS 1.29.x.
